core/src/types/buffer.rs

```diff
@@ -317,6 +317,47 @@ impl Buffer {
             }
         }
     }
+
+    /// Splits the current object into two parts at the specified index (`at`).
+    ///
+    /// The portion before the index is stored in the left tuple element, while the portion
+    /// after the index is stored in the right tuple element.
+    #[inline]
+    pub fn split(&self, at: usize) -> (Self, Self) {
```

Comment:
I expect we can align with `Bytes`'s [split](https://docs.rs/bytes/latest/bytes/struct.Bytes.html#method.split_to) API, would you like to take a look?

Reply:
The meaning is that internally, "split" actually calls "split_to"?

Reply:
> The meaning is that internally, "split" actually calls "split_to"?

No, I'm talking about the API design:

- `pub fn split_off(&mut self, at: usize) -> Self`
- `pub fn split_to(&mut self, at: usize) -> Self`

Maybe we don't need `split`?

Reply:
Yes, I initially thought of providing a 'split', returning two parts, left and right, so that developers can freely handle the data of these two parts.

Now I have removed this implementation.